modules/adding-ibmz-lpar-agent.adoc

```diff
@@ -0,0 +1,107 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_with_agent_based_installer/prepare-pxe-infra-agent.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="adding-ibmz-lpar-agents_{context}"]
+= Adding {ibm-z-title} Logical Partition (LPAR) as agents
+
+Use the following procedure to manually add {ibm-z-name} agents to your cluster that runs in an LPAR environment. Use this procedure only for {ibm-z-name} clusters running in an LPAR.
+
+.Procedure
+
+. Create a boot parameter file for the agents.
++
+.Sample parameter file:
+[source,terminal]
+----
+rd.neednet=1 cio_ignore=all,!condev \
+console=ttysclp0 \
+ignition.firstboot ignition.platform.id=metal
+coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img \// <1>
+coreos.inst.persistent-kargs=console=ttysclp0
+ip=<ip>::<gateway>:<netmask>:<hostname>::none nameserver=<dns> \// <2>
+rd.znet=qeth,<network_adaptor_range>,layer2=1
+rd.<disk_type>=<adapter> \// <3>
+zfcp.allow_lun_scan=0
+ai.ip_cfg_override=1 \// <4>
+random.trust_cpu=on rd.luks.options=discard
+----
+<1> For the `coreos.live.rootfs_url` artifact, specify the matching `rootfs` artifact for the `kernel` and `initramfs` that you are starting. Only HTTP and HTTPS protocols are supported.
+<2> For the `ip` parameter, manually assign the IP address, as described in _Installing a cluster with z/VM on IBM Z and IBM LinuxONE_. 
+<3> For installations on DASD-type disks, use `rd.dasd` to specify the DASD where Red Hat Enterprise Linux CoreOS (RHCOS) is to be installed. For installations on FCP-type disks, use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where RHCOS is to be installed.
+<4> Specify this parameter when you use an Open Systems Adapter (OSA) or HiperSockets.
+
+. Generate the `.ins` and `initrd.img.addrsize` files.
```

Comment:
I believe that this step needs to be rewritten in a way that's simpler for users to understand. It seems like this is really one large script, where all these commands can be run in a single batch. 

If that's the case, then I understand the attempt to describe what each group of commands is doing in a substep, but I believe that creates this weird situation where it looks like we're telling users to run a series of commands in a command line, but really we're expecting them to run everything at once. This creates an issue both in terms of style guide rules as well as the logical flow of the procedure.

So if all the commands in this step need to be run in a single script, I think the following changes need to be done:

1.  The commands in all of the substeps, 2a-2g, should be placed in one large code block so that everything is together, and it is made clear that this all needs to be run as a single script
2. If you still want to describe what the individual commands in the script are doing, then find some way to add that description. I think the most straightforward method would be to use callout labels like you do in step 1 of this procedure. There are other ways, such as including a paragraph before the code block to explain what is happening, but I think the callouts would look the best.

Reply:
Also, it would be best to explicitly mention in the procedure that users need to run a python script specifically:
```suggestion
. Generate the `.ins` and `initrd.img.addrsize` files by running the following Python script:
+
[source,terminal]
```